aten/src/ATen/native/mkldnn/Linear.cpp

```diff
@@ -24,22 +24,59 @@ Tensor mkldnn_linear(
 namespace at {
 namespace native {
 
+// Helper function for getting an ideep tensor out of an aten Tensor.
+// In case the aten Tensor is a mkldnn tensor,
+//   1. reshape first if input dim is greater than 2 and the reshape will cost a memory copy.
+//   2. directly return the ideep tensor if the input dim is 2.
+// In case the aten Tensor is a dense tensor, make sure:
+//   1. the aten tensor is 2d, so that 3d*2d Linear is viewed as 2d*2d.
+//   2. the aten tensor is contiguous.
+inline ideep::tensor get_mkldnn_tensor(const at::Tensor& tensor) {
```

Comment:
this function is used in a single place, so could be put there directly too

Reply:
yes, `get_mkldnn_tensor` is no longer needed.